sdk/state/payment.go

```diff
@@ -23,6 +23,7 @@ type CloseAgreementDetails struct {
 	ObservationPeriodLedgerGap int64
 	IterationNumber            int64
 	Balance                    int64
+	FinalPaymentAmount         int64
```

Comment:
@leighmcculloch fyi `LastPaymentAmount` that you suggested [here](https://github.com/stellar/experimental-payment-channels/issues/248#issuecomment-906815483) had me initially confused with the meaning of last as in the previous close agreement. So I changed to use Final instead

Reply:
I agree, naming for this is going to be hard.

`FinalPaymentAmount` also has an ambiguiity problem, because final payment amount sounds a lot like the net settlement amount, which is the final amount that will be exchanged between the participants when the channel closes.

![Untitled-2021-08-20-2126-2](https://user-images.githubusercontent.com/351529/131930620-8634d23f-73a8-4cd8-83b3-729fe43c7223.png)

Reply:
I agree, naming for this is going to be hard.

`FinalPaymentAmount` also has an ambiguiity problem, because final payment amount sounds a lot like the net settlement amount, which is the final amount that will be exchanged between the participants when the channel closes.

![Untitled-2021-08-20-2126-2](https://user-images.githubusercontent.com/351529/131930620-8634d23f-73a8-4cd8-83b3-729fe43c7223.png)

The names we've considered so far, and some other names:

- `LastPaymentAmount` – Problem is this sounds a lot like the previous close agreements payment rather than this close agreements payment.
- `FinalPaymentAmount` – Problem is this sounds a lot like the net-settlement payment when the channel closes.
- `PaymentAmount` – Problem it sounds like it could be the close agreement, or the individual payment amount.
- `ProposerPaymentAmount` 
- `BalanceDiff` – This could be an initiator relative amount, similar to how you implemented the relativeness of the amount in this PR. We could use this to calculate a proposer payment amount.
- `LastBalance`/`PreviousBalance` – What if we just encode the last balance. So the current balance and last balance are encoded. Theoretically we can calculate whatever else we want from that. The name for this field is ultra clear too.

Reply:
imo I like having one payment amount field and one balance field. Instead of two balance fields, which feels redundant since we'd most likely just be using it to get the payment amount anyways. `LastBalance` also has the same problem of using `Last` 😆 .

I initially liked `ProposePaymentAmount` but then if I looked at a close agreement 5 back I wouldn't remember if I proposed or not.

I think my top vote would be `PaymentAmount`, and allow it to be negative. It's consistent with how Balance is defined currently.

 I don't feel _super_ strongly about this though, so if you like 2 balance fields I'm game

Reply:
> `PaymentAmount`, and allow it to be negative

I think this will be confusing. When a proposer who is the responder proposes a payment they pass into the `ProposePayment` function a `amount` that is always positive, but when they receive `PaymentSentEvent` the `CloseAgreement` within will have `PaymentAmount` as negative. Now that I think of it, I think if we're going to have a field for the payment amount that mirrors the `amount` parameter of the `ProposePayment` function, we should make this field behave identically. Then there's consistency between the `ProposePayment(amount int64)` and the `CloseAgreement{PaymentAmount int64}`.

Wdyt?

> It's consistent with how Balance is defined currently.

That's true. I think the negative `Balance` is convenient for the internals, but it's a shame that it leaks outside via the `CloseAgreement` and we don't have to carry that to other fields if they don't need it.

Reply:
> I think if we're going to have a field for the payment amount that mirrors the amount parameter of the ProposePayment function, we should make this field behave identically

true. So how about `PaymentAmount` and leave it as always positive? And then the `ProposingSigner` field can tell who was the one who proposed instead of the positive vs negative

Reply:
Yup I think that works.

Reply:
https://github.com/stellar/experimental-payment-channels/pull/291/commits/c51561b26664f3693a2d9474eb85208e0e50bd78

Reply:
fyi added a helper method here for checking if the closeAgreement proposer is the initiator. That also led me to add a validation for checking that the ProposingSigner is either the local or remote signer, so that it's safe to use this helper method

https://github.com/stellar/experimental-payment-channels/pull/291/commits/ecb15c57dfa6e6a11e9585bd865930c52550682d